src/server/game/Handlers/MiscHandler.cpp

```diff
@@ -1638,6 +1637,15 @@ void WorldSession::HandleMoveSetCanFlyAckOpcode(WorldPacket& recv_data)
 
     recv_data.read_skip<float>();                           // unk2
 
+    if (movementInfo.HasMovementFlag(MOVEMENTFLAG_CAN_FLY))
+    {
+        sScriptMgr->AnticheatSetCanFlybyServer(_player, true);
+    }
+    else
+    {
+        sScriptMgr->AnticheatSetCanFlybyServer(_player, false);
+    }
```

Comment:
```suggestion
    sScriptMgr->AnticheatSetCanFlybyServer(_player, movementInfo.HasMovementFlag(MOVEMENTFLAG_CAN_FLY));
```

instead of only `movementInfo.HasMovementFlag(MOVEMENTFLAG_CAN_FLY)` you can also make a cast to boolean